docs/DeveloperGuide.adoc

```diff
@@ -341,6 +349,212 @@ Use case ends.
+4.  AddressBook adds new visitation record for chosen patient
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Edit visitation record
+
+*MSS*
+
+1.  User requests to edit visitation record for specific patient on specific date
+2.  AddressBook shows pop-up form for user to edit visitation record
+3.  User saves edit
+4.  AddressBook saves edit
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The record does not exist.
++
+[none]
+** 1a1. AddressBook shows an error message.
+
+Use case resumes at step 1.
+
+* 1b. There are multiple records with the same name and date
++
+[none]
+** 1b1. AddressBook shows list of records and prompts user to re-enter last command with index of record to edit.
+
+Use case resumes at step 1.
+
+[discrete]
+=== Use case: Delete visitation record
+
+*MSS*
+
+1.  User requests to delete visitation record for specific patient on specific date
+2.  AddressBook deletes visitation record
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The record does not exist.
++
+[none]
+** 1a1. AddressBook shows an error message.
+
+Use case resumes at step 1.
+
+* 1b. There are multiple records with the same name and date
++
+[none]
+** 1b1. AddressBook shows list of records and prompts user to re-enter last command with index of record to delete.
+
+Use case resumes at step 1.
+
+[discrete]
+=== Use case: Save user-defined macros
+
+*MSS*
+
+1.  User requests to save user-defined macros
+2.  AddressBook saves user-defined macros
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The content of the macro is empty.
++
+[none]
+** 1a1. AddressBook shows an error message.
++
+Use case resumes at step 1.
+
+[none]
+* 1b. The shortcut of the macro is illegal.
++
+[none]
+** 1b1. AddressBook shows an error message.
++
+Use case resumes at step 1.
+
+[discrete]
+=== Use case: Invoke user-defined macros
+
+*MSS*
+
+1.  User requests to invoke user-defined macros
+2.  AddressBook outputs user-defined data
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The command is not defined.
++
+[none]
+** 1a1. AddressBook shows an error message.
++
+Use case resumes at step 1.
+
+[discrete]
+=== Use case: View reminders
+
+*MSS*
+
+1.  User starts up AddressBook
+2.  Reminders are displayed
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2. No reminders are set to show
++
+[none]
+** 2a. AddressBook shows no reminder message.
++
+Use case ends.
+
+[discrete]
+=== Use case: View follow-up reminders
+
+*MSS*
+
+1.  User starts up AddressBook
+2.  Follow-up reminders for the day are displayed
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2. No reminders are scheduled for the day
++
+[none]
+** 2a. AddressBook shows no reminder message.
++
+Use case ends.
+
 _{More to be added}_
 
 [appendix]
```

Comment:
Some comments on your NFRs:

1. Try to update them soon! For example, we could rename persons to patients to make it more specific 😄 
1. Think of more specific NFRs tied to your application.